app/services/notifications_service.rb

```diff
@@ -0,0 +1,60 @@
+class NotificationsService
+
+  @@gcm = GCM.new(ENV['GOOGLE_API_KEY'])
+
+  def self.notify_users_on_trip(trip)
+    users = []
+    if trip.shopper.tripping?
+      users = User.idle.where.not(notification_key: nil)
+    elsif trip.shopper.idle?
+      # meaning trip was cancelled
+      # TODO: complete this
```

Comment:
You only need to decline all pending requests when a trip is cancelled (which shouldn't happen here anyway).